### ISSUE TITLE
[BUG] Operation not permitted (ix-file-permissions.sh)

### ISSUE THREAD

**Issue by varioous** [TYPO3-Solr/ext-solr#3837]
When starting solr service the start fails everytime with: 

`ddev logs -s solr
/opt/solr/docker/scripts/solr-foreground: running /docker-entrypoint-initdb.d/as-sudo/fix-file-permissions.sh
chown: changing ownership of '/var/solr/data/cores': Operation not permitted`

.ddev-docker-compose.solr.yaml
```yaml
services:
  solr:
    container_name: ddev-${DDEV_SITENAME}-solr
    image: typo3solr/ext-solr:12.0.x-dev
    ports:
      - 8983
    labels:
      com.ddev.site-name: ${DDEV_SITENAME}
      com.ddev.approot: $DDEV_APPROOT
    environment:
      - VIRTUAL_HOST=$DDEV_HOSTNAME
      - HTTP_EXPOSE=8983
    volumes:
      - ./solr/cores:/var/solr/data/cores
      - solr:/var/solr
  web:
    links:
      - solr

volumes:
  solr:
```

**Expected behavior**
Solr started successfully

**Used versions:**
 - TYPO3 Version:12.4
 - EXT:solr Version: 12.0.0

**Additional context**
Setup: Windows + Docker-Desktop + WSL2

**Comment by dkd-kaehm** [comment]
@varioous
The path `.ddev/solr/cores` is mounted to the exported docker volume without synchronize the unix user ids.
In Apache Solr and EXT:solr docker images, the service runs with unix user id 8983, your host user has other UID.
The circular bind mounts  `.ddev/solr/cores` on sub path of `exported docker volume` is not good idea.

If you want to enable/disable cores, use the provided feature ENV-"TYPO3_SOLR_ENABLED_CORES":
https://docs.typo3.org/p/apache-solr-for-typo3/solr/12.0/en-us/Appendix/DockerTweaks.html
If not, then just remove the `- ./solr/cores:/var/solr/data/cores` line.

If you want for example deploying the stuff from your git repo to Apache Solr Service/Stages,
or test some changes on Schemas, etc,
then inspire you from [our solr-ddev-site](https://github.com/TYPO3-Solr/solr-ddev-site/blob/8a59bcb9b24e964415e5c2375fa7917bd17941f1/.ddev/docker-compose.solr.yaml).

We'll not fix the "specials" like this, to avoid even more complexity, but we want to document the Docker Build tweak of EXT:solr image on #3838.

**Comment by itzonban** [comment]
Hi,

I also ran into this issue (switching from 11.5 to 12.0.5) and I think I found the cause of it:

In v11.6 you introduced the new files [here](https://github.com/TYPO3-Solr/ext-solr/tree/main/Docker/SolrServer/docker-entrypoint-initdb.d). The workflow is supposed to be that the `as-sudo-tweaks.sh` is called and then all scripts inside the folder `as-sudo` are executed with `sudo` permissions.

But in the solr Docker, which your image is based on, the [runinit-db](https://github.com/apache/solr/blob/main/solr/docker/scripts/run-initdb) script is looking for files to execute with this command:

```bash
find /docker-entrypoint-initdb.d/ -mindepth 1 -type f | sort -n
```

which in your case results in the following output order:

```bash
/docker-entrypoint-initdb.d/as-sudo/fix-file-permissions.sh
/docker-entrypoint-initdb.d/as-sudo-tweaks.sh
/docker-entrypoint-initdb.d/disable-cores.sh
```

So the `fix-file-permissions.sh` inside the `as sudo` folder will be executed first and this without `sudo` permission - this causes the `Operation not permitted` failure trying to change the file permissions and aborts the container. By either renaming the `as-sudo` folder or the `as-sudo-tweaks.sh` to offer the correct file order in the sorting this issue should be fixed.

Best regards
Daniel
